### PR TITLE
YONK-747: Oauth support for course overview API

### PR DESCRIPTION
**Pull request by aamishbaloch**
Changes include Oauth support for course overview API.

### CODE REVIEW

**Comment by aamishbaloch** [comment]
@ziafazal Can you review this?

File: edx_solutions_api_integration/mobile_api/tests/test_courses.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    def test_with_invalid_username(self):
+        """
+        Test scenario when requested user and logged in user are not same
+        """
+        self.login()
+
+        response = self.api_response(expected_response_code=None, data={'username': 'other_user'})
+        self.assertEqual(response.status_code, 403)
+
+    def test_user_invalid_course(self):
```
Comment (ziafazal) [inline edx_solutions_api_integration/mobile_api/tests/test_courses.py]: nit: can we rename this test to `test_not_enrolled_user`

File: edx_solutions_api_integration/permissions.py [hunk 2]
```diff
@@ -86,6 +87,23 @@ def has_permission(self, request, view):
             return True
 
 
+class IsStaffOrEnrolled(permissions.BasePermission):
+    """
+    Permission that allows access to admin users or the enrolled users of a course.
```
Comment (ziafazal) [inline edx_solutions_api_integration/permissions.py]: nit replace `admin` in this line with `staff`

File: edx_solutions_api_integration/permissions.py [hunk 3]
```diff
@@ -86,6 +87,23 @@ def has_permission(self, request, view):
...
+    """
+    def has_permission(self, request, view):
+        user = request.user
+        course_id = request.parser_context.get('kwargs', {}).get('course_id', None)
+
+        if course_id:
+            course_key = get_course_key(course_id)
```
Comment (ziafazal) [inline edx_solutions_api_integration/permissions.py]: Can we move this statement `course_key = get_course_key(course_id)`
above `If` statement and then only check `if course_key`

**Comment by aamishbaloch** [comment]
@ziafazal Done with the suggested changes. Can you review this?

**Pull request merged**